### ISSUE TITLE
Investigate replacing Babel with TypeScript compiler 

### ISSUE THREAD

**Issue by ZhangYiJiang** [nusmodifications/nusmods#1550]
Since we've moved to TypeScript, we may want to try to use `tsc` not just as a type checker but also for transpilation, by replacing Babel and using `ts-loader` in Webpack  

### Possible advantages

- Smaller code because by having type information TSC may be able to produce more optimal JS 
- Less config - instead of having to configure TS and Babel separately we now only have TS to configure 

### Possible disadvantages 

- Less flexible - it seems like TSC doesn't have the fine grained browser targeting that Bebel's `present-env` allows, which means its output may be less optimal 

We should try setting this up on a test branch and compare the output

**Comment by li-kai** [comment]
I don't think this will be in our focus in any time, due to TSC's inability to target browser runtimes. This issue will be here for a while.
